src/almanack/metrics/data.py

```diff
@@ -770,3 +776,57 @@ def get_github_build_metrics(
 
     # else we return an empty dictionary
     return {}
+
+
+def get_ecosystems_package_metrics(repo_url: str) -> Dict[str, Any]:
+    """
+    Fetches package data from the ecosystem API and calculates metrics
+    about the number of unique ecosystems, total version counts,
+    and the list of ecosystem names.
+
+    Args:
+        repo_url (str):
+            The repository URL of the package to query.
+
+    Returns:
+        Dict[str, Any]:
+            A dictionary containing information about packages
+            related to the repository.
+    """
+
+    if repo_url is None:
+        LOGGER.warning(
+            "Did not receive a valid repository URL and unable to gather package metrics."
+        )
+        return {}
+
+    # normalize http to https if necessary
+    if repo_url.startswith("http://"):
+        LOGGER.warning(
+            "Received `http://` repository URL for package metrics search. Normalizing to use `https://`."
+        )
+        repo_url = repo_url.replace("http://", "https://")
+
+    # perform package srequest
+    package_data = get_api_data(
+        api_endpoint="https://packages.ecosyste.ms/api/v1/packages/lookup",
+        params={"repository_url": repo_url},
+    )
```

Comment:
Nice to see the warning messages you added above and the fix to translate it into HTTPS.

This call to `get_api_data()` is the one I'm mostly concerned about; if it's given a URL like `git@github.com:dangoslen/changelog-enforcer.git`, which would happen if someone cloned the repo via SSH, the API returns an empty list. It might help someone trying to debug the problem if you checked whether `package_data` is an empty list and emitted a warning like `f"Got an empty response from API for URL {repo_url}"`, in which case the user might see it was the wrong type of URL and might be able to fix it.

Alternatively, you could detect when the repo URL is `git@github.com:<user>/<repo>.git` and map it to `http://github.com/<user>/<repo>`, like how you did with the HTTP->HTTPS fix. It might be a good idea to do that anyway since I imagine this won't be an uncommon source of errors, and still have some logging when the API doesn't produce a response.

Reply:
Thanks for this! I hadn't thought about the SSH / `.git` URL's! I'll update this to do as you mentioned, attempting to convert the path to an HTTPS link.